src/functional/ffront/func_to_foast.py

```diff
@@ -93,8 +94,21 @@ def _preprocess_definition_ast(cls, definition_ast: ast.AST) -> ast.AST:
         return ucc
 
     @classmethod
-    def _postprocess_dialect_ast(cls, dialect_ast: foast.FieldOperator) -> foast.FieldOperator:
-        return FieldOperatorTypeDeduction.apply(dialect_ast)
+    def _postprocess_dialect_ast(
+        cls, dialect_ast: foast.FieldOperator, captured_vars: CapturedVars
+    ) -> foast.FieldOperator:
+        # ensure annotated matches with deduced return type
```

Comment:
```suggestion
        typed_dialact_ast = FieldOperatorTypeDeduction.apply(dialect_ast)
        # ensure annotated matches with deduced return type
```
Actually you can rename `dialect_ast` to `foast_node` everywhere here.